x-pack/plugins/aiops/public/components/log_categorization/format_category.tsx

```diff
@@ -0,0 +1,168 @@
+      const localEndOFToken = j + token.length + 1;
+      positions[i].start = offset + j + 1;
+      positions[i].end = offset + localEndOFToken;
+      // slice the example string to remove the token and preceding text
+      // to ensure we don't match future tokens in earlier text
+      tempExample = tempExample.slice(localEndOFToken);
+      offset += localEndOFToken;
+    }
+
+    tempExample = ` ${example} `;
+
+    // build up the list ot elements by chopping up the example string
+    // using the token positions found above
+    const elements: JSX.Element[] = [];
+    let pos = 0;
+    for (let i = 0; i < positions.length; i++) {
+      elements.push(
+        <span css={wildcardStyle}>{tempExample.substring(pos, positions[i].start)}</span>
+      );
+
+      elements.push(
+        <span css={tokenStyle}>{tempExample.substring(positions[i].start, positions[i].end)}</span>
+      );
+      pos = positions[i].end;
+    }
+
+    elements.push(
+      <span css={wildcardStyle}>{tempExample.substring(positions[positions.length - 1].end)}</span>
+    );
+
+    return elements;
+  }
+
+  return createFormattedExample;
+};
+
+export const FormattedPatternExamples: FC<Props> = ({ category, count }) => {
+  const createFormattedExample = useCreateFormattedExample();
+
+  const e = useMemo(() => {
+    const { key, examples } = category;
+    const tempCount = count === undefined || count > examples.length ? examples.length : count;
+    const formattedExamples = new Array(tempCount)
+      .fill(0)
+      .map((_, i) => createFormattedExample(key, examples[i]));
+    return formattedExamples.map((example, i) => (
+      <>
+        <code>{example}</code>
+        {i < formattedExamples.length - 1 ? <EuiHorizontalRule margin="s" /> : null}
+      </>
+    ));
+  }, [category, count, createFormattedExample]);
+
+  return <WrapInText>{e}</WrapInText>;
+};
+
+export const FormattedRegex: FC<Props> = ({ category }) => {
+  const { tokenStyle, wildcardStyle } = useStyles();
+
+  const { regex } = category;
+  const formattedRegex = useMemo(() => {
+    const regexTokens = regex.split(/(\.\*\?)|(\.\+\?)/).filter((d) => d !== undefined);
+    const elements: JSX.Element[] = [];
+    for (let i = 0; i < regexTokens.length; i++) {
+      const token = regexTokens[i];
+      if (token.match(/\.\*\?|\.\+\?/)) {
+        elements.push(<span css={wildcardStyle}>{token}</span>);
+      } else {
+        elements.push(<span css={tokenStyle}>{token}</span>);
+      }
+    }
+    return elements;
+  }, [regex, tokenStyle, wildcardStyle]);
+  return (
+    <WrapInText>
+      <code>{formattedRegex}</code>
+    </WrapInText>
+  );
+};
+
+export const FormattedTokens: FC<Props> = ({ category }) => {
+  const { tokenStyle } = useStyles();
+
+  return (
+    <WrapInText>
+      <code css={tokenStyle}>{category.key}</code>
+    </WrapInText>
+  );
+};
+
+const WrapInText: FC = ({ children }) => (
+  <EuiText css={{ fontWeight: 'bold' }} size="s">
```

Comment:
Nit: This css could be broken to a definition outside the component scope. AFAIK the benefit would be that internally this would then be treated as a css class instead of inline css on every element.